src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -452,21 +452,30 @@ void DistributedSearchTreeImpl<DeviceType>::queryDispatch(
   ////////////////////////////////////////////////////////////////////////////
 }
 
-// FIXME: for some reason Kokkos::BinSort::sort() was not const.
-// If https://github.com/kokkos/kokkos/pull/1310 makes it into master in
-// Trilinos, we might want to pass bin_sort by const reference.
-template <typename BinSort>
-void applyPermutations(BinSort &)
+template <typename PermutationView>
+void applyPermutations(PermutationView const &)
 {
   // do nothing
 }
 
-template <typename BinSort, typename View, typename... OtherViews>
-void applyPermutations(BinSort &bin_sort, View view, OtherViews... other_views)
+template <typename PermutationView, typename View, typename... OtherViews>
+void applyPermutations(PermutationView const &permutation, View &view,
+                       OtherViews &... other_views)
 {
-  ARBORX_ASSERT(bin_sort.get_permute_vector().extent(0) == view.extent(0));
-  bin_sort.sort(view);
-  applyPermutations(bin_sort, other_views...);
+  static_assert(std::is_integral<typename PermutationView::value_type>::value,
+                "");
+  ARBORX_ASSERT(permutation.extent(0) == view.extent(0));
+  Kokkos::View<typename View::value_type *, typename View::device_type,
+               Kokkos::MemoryTraits<Kokkos::RandomAccess>>
+      scratch(Kokkos::ViewAllocateWithoutInitializing(view.label()),
+              view.size());
+  Kokkos::parallel_for(
+      ARBORX_MARK_REGION("permute"),
+      Kokkos::RangePolicy<typename View::execution_space>(0, view.size()),
+      KOKKOS_LAMBDA(int i) { scratch(i) = view(permutation(i)); });
```

Comment:
This assume that `view` is of the type `View<value_type*>`. This breaks DTK where we have `View::<value_type**>`
```
error: no match for call to ‘(const Kokkos::View<double**, Kokkos::Device<Kokkos::Serial, Kokkos::HostS
pace> >) (long unsigned int&)’
       KOKKOS_LAMBDA(int i) { scratch(i) = view(permutation(i)); });
                                           ~~~~^~~~~~~~~~~~~~~~
```

Reply:
We only call this function (and `sortResults`) in `ArborX_DetailsDistributedSearchTreeImpl.hpp` and it resides in the same file. Hence, `DTK` must call this function through a different code path (outside of `ArborX`) and thus relies on implementation details?

I can look into fixing this.

Reply:
Because ArborX was part of DTK, we call several functions that are in `Details`. I have a branch where I keep the changes needed to compile DTK with the latest version of  ArborX... For this error the call in DTK is [here](https://github.com/ORNL-CEES/DataTransferKit/blob/285d708d31c3b72a9e4a02f51352452002f7bff4/packages/Discretization/src/DTK_Interpolation_decl.hpp#L195)